packages/workbox-core/src/_private/DBWrapper.ts

```diff
@@ -266,7 +266,9 @@ export class DBWrapper {
     ...args: any[]
   ) {
     const callback = (txn: IDBTransaction, done: Function) => {
-      const objStore = txn.objectStore(storeName)
+      const objStore = txn.objectStore(storeName);
+      // TODO(philipwalton): Fix this underlying TS2684 error.
+      // @ts-ignore
```

Comment:
This is the biggest issue, and I just punted on it as I don't have any idea how to resolve it. @philipwalton, if you can look into it as part of either another PR or push an update to this PR, that would be great.